### PR TITLE
Initialize python backend

### PR DESCRIPTION
**Pull request by xxdunedainxx**
Issues included in this PR: 

https://github.com/xxdunedainxx/lindsay-wildlife-treasure-hunt/issues/14
https://github.com/xxdunedainxx/lindsay-wildlife-treasure-hunt/issues/12
https://github.com/xxdunedainxx/lindsay-wildlife-treasure-hunt/issues/11
https://github.com/xxdunedainxx/lindsay-wildlife-treasure-hunt/issues/4

Initializing python backend with some common code libraries i typically use + starting up the SMTP & QR code libs

### CODE REVIEW

**Review by rjmacleod — APPROVED** [review]
I glanced at everything, most of it is over my head but that's not too surprising. LGTM

File: backend/src/Configuration.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import json
+
+class Configuration:
+  DEFAULT_VALUES : dict = {
+    "SMTP_SERVER" : "smtp.gmail.com",
+    "SMTP_PORT"   : 465
+  }
+
+  def __init__(self):
+    self.CONF_FILE_LOCATION = './conf.json' if 'SYNCER_CONF' not in os.environ.keys() else os.environ['SYNCER_CONF']
```
Comment (xxdunedainxx) [inline backend/src/Configuration.py:11]: set self.CONF_FILE_LOCATION: str 
```suggestion
    self.CONF_FILE_LOCATION: str = './conf.json' if 'SYNCER_CONF' not in os.environ.keys() else os.environ['SYNCER_CONF']
```

File: backend/src/QRCode/QRCodeGenerator.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import qrcode
+from src.util.LogFactory import LogFactory
+from src.util.FileIO import FileIO
+import os
+
+class QRCodeGenerator:
+  QRCODE_DIRECTORY: str = './tmp'
+
+  @staticmethod
+  def generate_qr_code_file(address: str, outFile: str):
```
Comment (xxdunedainxx) [inline backend/src/QRCode/QRCodeGenerator.py:10]:
```suggestion
  def generate_qr_code_file(string: str, outFile: str):
```

File: backend/src/QRCode/QRCodeGenerator.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    fullPath: str = f"{QRCodeGenerator.QRCODE_DIRECTORY}{os.sep}{outFile}"
+
+    LogFactory.MAIN_LOG.info(f"Outputing address {address} to file {fullPath}")
+    if FileIO.file_exists(fullPath) == False:
+      LogFactory.MAIN_LOG.info(f"creating dir / file {fullPath}")
+      FileIO.create_file_if_does_not_exist(path=fullPath)
```
Comment (xxdunedainxx) [inline backend/src/QRCode/QRCodeGenerator.py:16]: need to test

File: backend/src/QRCode/QRCodeGenerator.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    img = qrcode.make(address)
+    img.save(outFile)
```
Comment (xxdunedainxx) [inline backend/src/QRCode/QRCodeGenerator.py:19]: fullPath 
```suggestion
    img.save(fullPath)
```

**Pull request merged**